### ISSUE TITLE
Benchmarking Cifar10

### ISSUE THREAD

**Issue by saisubramaniam147** [xyutao/bocl#2]
Hi again,

I trained session S1 of cifar10 on ResNet18, using SGD optimizer for 20 epochs with the learning rate provided in the paper. I am able to obtain only 45% at max. Can you please suggest anything I might have missed?

**Comment by saisubramaniam147** [comment]
Just to clarify, each session has 50,000/5 = 10,000 samples, or is it 50,000 each session but different settings of brightness and saturation are applied ?

**Comment by xyutao** [comment]
@saisubramaniam147 Each session has 10,000 samples. The learning rate used in each session can vary with different code implementations, so it is better to turn your own learning rates and training epochs for more considerable performances. 

_Trying a different initial learning rate (e.g., 0.1) and more epochs?_

**Comment by saisubramaniam147** [comment]
Thanks @xyutao, can you upload the ResNet code for this experiment? It will help me to benchmark in the same approach.

**Comment by xyutao** [comment]
OK, I'll upload necessary codes tomorrow.

**Comment by xyutao** [comment]
Hi @saisubramaniam147 , I've uploaded several files to support benchmarking. Try running:

> bash script/cifar10/full/cifar10_resnet20_v1_naive.sh

The script loads a pre-stored snapshot trained at session 1 at the "params/cifar10_baseline" folder. The training will begin from session 1's model (test acc. 67.05%) and continuously learn the remaining 4 sessions. You may tune the hyper-parameters defined in the shell scripts to get different performances.

If you want to train session 1 from scratch, remove the **--resume-s1** and **--resume-from** options, and tune the **--epochs**, **--lr**, **--lr-decay-epoch** hyper-parameters. For example, you may set a larger epoch and decay steps to get a  base model with higher accuracy. But be sure to make all comparable methods start from the same session-1 model for fair comparison.

You may notice that the accuracy could drop at some sessions and then increase again. This is influenced by the order of the training sessions and the order of the applied illumination&saturation . For example, change the code in data/datasets.py L40& L41:
`self._brightness_jitter = [0, -0.1, 0.1, -0.2, 0.2]` 
`self._saturation_jitter = [0, -0.1, 0.1, -0.2, 0.2]`
to a different order:
`self._brightness_jitter = [0, -0.2, -0.1, 0.1, 0.2]` 
`self._saturation_jitter = [0, -0.2, -0.1, 0.1, 0.2]`
You can get a more smooth changing of the test accuracies. So, it is recommended to try different sequential orders of the training sessions or illumination&saturation and average the results.

**Comment by saisubramaniam147** [comment]
Thanks for uploading the code.

As you rightly mentioned, the accuracy drops at sessions where the brightness and saturation changes from positive to negative or viz. Can you share the exact order and values of _brightness_jitter and _saturation_jitter, as reported in the paper, where the accuracies increase without dropping in Fig 3 a and c?

**Comment by xyutao** [comment]
Adding the new patterns smoothly:

`self._brightness_jitter=[0, -0.2, -0.1, 0.1, 0.2]`
`self._saturation_jitter=[0, 0.1, 0.2, -0.2, -0.1]`

and then the accuracies should be:

> 67.09, 70.02, 71.95, 72.48, 74.00

**Comment by saisubramaniam147** [comment]
The above pattern works fine :) One final help - possible to give the numbers from the plot? It will be more accurate than me trying to guess the decimal points from the Figure.

Thanks again!
